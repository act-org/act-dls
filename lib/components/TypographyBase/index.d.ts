/**
 * @prettier
 */
import * as React from 'react';
import { TypographyProps } from '@material-ui/core';
export declare type Props = TypographyProps;
declare const TypographyBase: React.FC<Props>;
export default TypographyBase;
//# sourceMappingURL=index.d.ts.map