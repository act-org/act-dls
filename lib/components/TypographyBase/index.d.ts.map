{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/TypographyBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAc,eAAe,EAAE,MAAM,mBAAmB,CAAC;AAEhE,oBAAY,KAAK,GAAG,eAAe,CAAC;AAEpC,QAAA,MAAM,cAAc,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAEmB,CAAC;AAExD,eAAe,cAAc,CAAC"}