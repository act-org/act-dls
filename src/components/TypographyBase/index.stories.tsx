/**
 * @prettier
 */

import * as React from 'react';
import { select, text } from '@storybook/addon-knobs';

import TypographyBase from '.';

export const Base = (): React.ReactElement<any> => (
  <TypographyBase
    color={select(
      'Color',
      {
        Error: 'error',
        Inherit: 'inherit',
        Initial: 'initial',
        Primary: 'primary',
        Secondary: 'secondary',
        TextPrimary: 'textPrimary',
        TextSecondary: 'textSecondary',
      },
      'initial',
    )}
    variant={select(
      'Variant',
      {
        Body1: 'body1',
        Body2: 'body2',
        Button: 'button',
        Caption: 'caption',
        H1: 'h1',
        H2: 'h2',
        H3: 'h3',
        H4: 'h4',
        H5: 'h5',
        H6: 'h6',
        Inherit: 'inherit',
        Overline: 'overline',
        SrOnly: 'srOnly',
        Subtitle1: 'subtitle1',
        Subtitle2: 'subtitle2',
      },
      'body1',
    )}
  >
    {text('Text', 'Typography')}
  </TypographyBase>
);

export default {
  component: TypographyBase,
  parameters: {
    info: {
      text: 'This is the base variant of the Typography component.',
    },
  },
  title: 'Atoms/Typography',
};
