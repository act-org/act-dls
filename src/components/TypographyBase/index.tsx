/**
 * @prettier
 */

import * as React from 'react';
import { Typography, TypographyProps } from '@material-ui/core';

export type Props = TypographyProps;

const TypographyBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <Typography {...props} />;

export default TypographyBase;
