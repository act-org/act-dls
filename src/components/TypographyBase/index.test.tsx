/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import TypographyBase from '.';

describe('TypographyBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <TypographyBase variant="body1">This is a test.</TypographyBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
