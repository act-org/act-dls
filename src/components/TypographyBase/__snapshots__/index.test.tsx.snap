// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypographyBase matches the snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    This is a test.
  </p>
</div>
`;
